Winleafs.Wpf/Views/Layout/CreateEffectWindow.xaml.cs

```diff
@@ -0,0 +1,262 @@
+﻿using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Media;
+using Winleafs.Models.Models;
+using Winleafs.Models.Models.Layouts;
+using Winleafs.Wpf.Api;
+using Winleafs.Wpf.Api.Effects;
+using Winleafs.Wpf.Api.Layouts;
+using Winleafs.Wpf.Helpers;
+using Winleafs.Wpf.Views.Popup;
+
+namespace Winleafs.Wpf.Views.Layout
+{
+
+	/// <summary>
+	/// Interaction logic for CreateEffectWindow.xaml
+	/// </summary>
+	public partial class CreateEffectWindow : Window
+    {
+
+		private IList<FrameListItem> _frameListItems;
+		private CustomEffect _customEffect;
+		private Frame _currentFrame;
+		private Dictionary<uint, SolidColorBrush> _rgbToBrushMap;
+		private static readonly Regex _numericRegex = new Regex("[^0-9.]");
+		private static readonly string _frameNameFormat = Layout.Resources.Frame + " {0}";
+
+		public CreateEffectWindow()
+        {
+            InitializeComponent();
+
+			LayoutDisplay.EnableClick();
+			LayoutDisplay.InitializeResizeTimer();
+			LayoutDisplay.DrawLayout();
+            LayoutDisplay.DisableColorTimer();
+			LayoutDisplay.MultiSelectEnabled = false;
+			LayoutDisplay.PanelClicked += LayoutDisplay_PanelClicked;
+			ColorPicker.SelectedColor = Colors.White;
+
+            if (UserSettings.Settings.ActiveDevice.CustomEffect != null)
+            {
+                var serialized = JsonConvert.SerializeObject(UserSettings.Settings.ActiveDevice.CustomEffect); //Deep copy the custom effect when editing
+                _customEffect = JsonConvert.DeserializeObject<CustomEffect>(serialized);
+			}
+            else
+            {
+                _customEffect = new CustomEffect();
+			}
+
+			BuildRgbToBrushMap();
+
+			BuildFrameList();
+		}
+				
+		private void LayoutDisplay_PanelClicked(object sender, System.EventArgs e)
+		{
+			var drawablePanel = sender as DrawablePanel;
+
+			if (drawablePanel != null)
```

Comment:
Check if its null instead and return if that's the case.

Reply:
Actually even better would be turning this into an is cast like

```
			if (!(drawablePanel is DrawablePanel drawablePanel))
```

I'm typing this from memory so the syntax might be slightly incorrect but its a nice syntax

Reply:
Returning after the check is done would still be preferred even with this `is` cast

Reply:
Ah, I was forgetting that new syntax! I'll take a look at that - nice.